src/external/Proxy.sol

```diff
@@ -0,0 +1,61 @@
+pragma solidity 0.8.19;
+
+import {TransparentUpgradeableProxy} from "@openzeppelin-contracts/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import {IERC20} from "@openzeppelin-contracts/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin-contracts/contracts/token/ERC20/utils/SafeERC20.sol";
+
+contract Proxy is TransparentUpgradeableProxy {
+    using SafeERC20 for IERC20;
+
+    /// logic contract on testnet: 0x10e5196375e6461703662DE1c11F0C28884cBf3b
+
+    // address owner,
+    // address morpho,
+    // uint256 initialTimelock,
+    // address _asset,
+    // string memory _name,
+    // string memory _symbol
+    constructor(
+        address _morpho,
+        address _logic,
+        address _admin,
+        address _owner,
+        uint256 _timelock,
+        address _underlyingAsset,
+        string memory _name,
+        string memory _symbol
+    ) TransparentUpgradeableProxy(_logic, _admin, "") {
+        IERC20(_underlyingAsset).safeApprove(_morpho, type(uint256).max);
+
+        require(bytes(_name).length <= 30, "name too long");
```

Comment:
I think this can be upto 31 bytes. [Ref](https://docs.soliditylang.org/en/latest/internals/layout_in_storage.html#bytes-and-string).